tempest/api/workloadmgr/network/test_network_restore.py

```diff
@@ -325,61 +131,142 @@ def test_network_restore_cli(self):
+                    tvaultconf.PASS)
+                LOG.debug("Command executed correctly")
 
-            if rt_bf == rt_af:
+            self.wait_for_snapshot_tobe_available(workload_id, snapshot_id)
+            restore_id = query_data.get_snapshot_restore_id(snapshot_id)
+            if(self.getRestoreStatus(workload_id, snapshot_id, restore_id) == "available"):
                 reporting.add_test_step(
-                    "Verify router details after network restore",
-                    tvaultconf.PASS)
+                    "Network topology restore from CLI", tvaultconf.PASS)
             else:
                 reporting.add_test_step(
-                    "Verify router details after network restore",
-                    tvaultconf.FAIL)
-                LOG.error(
-                    "Router details before and after restore: {0}, {1}".format(
-                        rt_bf, rt_af))
+                    "Network topology restore from CLI", tvaultconf.FAIL)
+                raise Exception(
+                    "Network topology restore from CLI failed")
 
+            nt_af, sbnt_af, rt_af, intf_af = self.get_topology_details()
             LOG.debug(
                     "Interface details before and after restore: {0}, {1}".format(
-                        intf_bf, intf_af))
+                        intf_bf_1, intf_af))
 
+            vm_details_af = {}
             restored_vms = self.get_restored_vm_list(restore_id)
-            LOG.debug(f"Restored vms: {restored_vms}")
-            if len(restored_vms) == 0:
-                reporting.add_test_step("Instances not restored",
-                        tvaultconf.PASS)
-            else:
-                reporting.add_test_step("Instances are restored",
-                        tvaultconf.FAIL)
-                LOG.error("Instances are restored")
+            for vm in restored_vms:
+                vm_details = self.get_vm_details(vm)['server']
+                vm_details_af[vm_details['name'].replace(
+                    'restored_instance', '')] = vm_details
+
+            self.verify_network_restore(nt_bf_1, nt_af, sbnt_bf_1, sbnt_af,
+                    rt_bf_1, rt_af, vm_details_bf_1, vm_details_af,
+                    test_type='CLI')
+
+            for rvm in restored_vms:
+                self.delete_vm(rvm)
             self.delete_network_topology()
 
         except Exception as e:
             LOG.error("Exception: " + str(e))
+            reporting.add_test_step(str(e), tvaultconf.FAIL)
             reporting.set_test_script_status(tvaultconf.FAIL)
         finally:
             reporting.test_case_to_write()
 
+    def test_5_cleanup(self):
+        try:
+            for snapshot_id in snapshot_ids:
+                self.addCleanup(self.snapshot_delete, workload_id, snapshot_id)
+            self.addCleanup(self.workload_delete, workload_id)
+        except Exception as e:
+            LOG.error("Exception: " + str(e))

```

Comment:
Do we need to add test_script_status to FAIL in case if snapshot/workload cleanup functionality fails in addCleanup() ?

Reply:
snapshot & workload deletion is already validated as part of separate tests.